libs/feature/search/src/lib/elasticsearch/mapper/elasticsearch.mapper.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core'
+import { RecordSummary } from '@geonetwork-ui/util/shared'
+import { ElasticsearchFieldMapper } from './elasticsearch.field.mapper'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ElasticsearchMapper {
+  constructor(private fieldMapper: ElasticsearchFieldMapper) {}
+
+  toRecords(response: any): RecordSummary[] {
+    return response.hits.hits.map((hit) => this.toRecord(hit))
+  }
+
+  toRecord(hit: any) {
+    const { _source } = hit
+    const record: Partial<RecordSummary> = {
+      uuid: hit._id,
+      viewable: hit.view,
+      downloadable: hit.download,
+    }
+
+    Object.keys(_source).forEach((fieldName) => {
+      const mappingFn = (
+        this.getMappingFn(fieldName) || this.fieldMapper.mapGenericField
+      ).bind(this.fieldMapper)
+      mappingFn(record, _source, fieldName)
+    })
+
+    return record as RecordSummary
+  }
+
+  private getMappingFn(fieldName: string) {
+    return this.fieldMapper[`map${this.captitalizeFirstLetter(fieldName)}`]
```

Comment:
I'm a bit uncomfortable with using indirection for a method name. This often leads to code that is harder to search. For example with this system, if I look for `mapResourceTemporalExtentDateRange` globally in the project I will never find explicit usages even though it is actually used.

How about using a dict for the field mappers in `ElasticsearchFieldMapper`? The key would be the full, untransformed field name and the value is the mapper.

Reply:
> How about using a dict for the field mappers in `ElasticsearchFieldMapper`? The key would be the full, untransformed field name and the value is the mapper.

Yes I agree it's more readable, thanks !